src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,119 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
 public class ArrayList<T> implements List<T> {
 
+    private static final int DEFAULT_CAPACITY = 16;
+
+    private T[] arrayList;
+    private int filledWithElements = 0;
+
+    public ArrayList() {
+        arrayList = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
+    public ArrayList(int initialCapacity) {
+        arrayList = (T[]) new Object[initialCapacity];
+    }
+
+    public void resize(int newLength) {
+        T[] newArrayList = (T[]) new Object[newLength];
+        System.arraycopy(arrayList, 0, newArrayList, 0, filledWithElements);
+        arrayList = newArrayList;
+    }
+
     @Override
     public void add(T value) {
-
+        if (filledWithElements == arrayList.length) {
+            resize((int) (arrayList.length * 1.5));
+        }
+        arrayList[filledWithElements++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > filledWithElements) {
+            throw new ArrayIndexOutOfBoundsException();
+        } else if (filledWithElements == arrayList.length) {
+            resize((int) (arrayList.length * 1.5));
+        }
+        System.arraycopy(arrayList, index, arrayList, index + 1, filledWithElements - index);
+        arrayList[index] = value;
+        filledWithElements++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        while (arrayList.length < filledWithElements + list.size()) {
+            resize((int) (arrayList.length * 1.5));
```

Comment:
you are always increasing your list by 1.5. Do you need to pass any parameters at all to this method?

Reply:
done